python/ray/tune/automl/search_policy.py

```diff
@@ -0,0 +1,229 @@
+
+    It works in a round-by-round way. For each experiment round,
+    it generates a bunch of parameter config permutations, submits
+    and keeps track of them. Once all of them finish, results will
+    be fed back to the algorithm as a whole.
+    """
+
+    CONTINUE = "CONTINUE"
+    TERMINATE = "TERMINATE"
+
+    def __init__(self, search_space, reward_attr):
+        """Initialize AutoMLSearcher.
+
+        Arguments:
+            search_space (SearchSpace): The space to search.
+            reward_attr: The attribute name of the reward in the result.
+        """
+        # Pass experiment later to allow construction without this parameter
+        super(AutoMLSearcher, self).__init__()
+
+        self.search_space = search_space
+        self.reward_attr = reward_attr
+
+        self.experiment = None
+        self.summary = None
+        self.best_trial = None
+        self._is_finished = False
+        self._parser = make_parser()
+        self._undone_count = 0
+        self._running_trials = {}
+        self._completed_trials = {}
+
+        self._iteration = 0
+        self._total_trial_num = 0
+        self._start_ts = 0
+
+    def setup(self, exp, summary):
+        self.experiment = exp
+        self.summary = summary
+
+    def get_best_trial(self):
+        """Returns the Trial object with the best reward_attr"""
+        return self.best_trial
+
+    def next_trials(self):
+        if self._undone_count > 0:
+            # Last round not finished
+            return []
+
+        trials = []
+        raw_param_list, extra_arg_list = self._select()
+        if not extra_arg_list:
+            extra_arg_list = [None] * len(raw_param_list)
+
+        for param_config, extra_arg in zip(raw_param_list, extra_arg_list):
+            tag = ''
+            new_spec = copy.deepcopy(self.experiment.spec)
+            for path, value in param_config.items():
+                tag += '%s=%s-' % (path.split('.')[-1], value)
+                merge_config(path, value, new_spec['config'])
+
+            trial = create_trial_from_spec(
+                new_spec, self.experiment.name,
+                self._parser, experiment_tag=tag
+            )
+
+            # AutoML specific fields set in Trial
+            trial.results = []
+            trial.best_result = None
+            trial.param_config = param_config
+            trial.extra_arg = extra_arg
+
+            trials.append(trial)
+            self._running_trials[trial.trial_id] = trial
+            # self.summary.on_trial_create(trial)
+
+        ntrial = len(trials)
+        self._iteration += 1
+        self._undone_count = ntrial
+        self._total_trial_num += ntrial
+        self._start_ts = time.time()
+        logging.info(("=========== BEGIN Experiment-Round: %s "
+                      "[%s NEW | %s TOTAL] ===========")
+                     % (self._iteration, ntrial, self._total_trial_num))
+        return trials
+
+    def _trial_add_result(self, trial, result):
```

Comment:
this doesn't need to be a separate function right?

Reply:
Done